packages/api/src/routes/medical/patient.ts

```diff
@@ -430,4 +438,61 @@ router.get(
   })
 );
 
+/** ---------------------------------------------------------------------------
+ * PUT /patient/:id/opting-out
+ *
+ * Returns whether the patient is opted out of data pulling and sharing.
+ *
+ * @param req.cxId The customer ID.
+ * @param req.param.patientId The ID of the patient whose data is to be returned.
+ * @param req.query.optingOut Boolean value to opt patient out or in.
+ */
+router.put(
+  "/opting-out",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const { cxId, patient } = getPatientInfoOrFail(req);
+    const optingOut = isTrue(getFrom("query").orFail("optingOut", req));
+
+    const result = await optingPatientInOrOut({ cxId, patientId: patient.id, optingOut });
+
+    const respPayload: PatientOptOutResponse = {
+      id: result.id,
+      optingOut: result.optingOut ?? false,
+      message: `Patient has been opted ${
+        result.optingOut ? "out of" : "in for"
+      } data pulling and sharing`,
+    };
+
+    return res.json(respPayload);
+  })
+);
+
+// TODO #2475 expose this on the patient
```

Comment:
This is only for a specific customer - once its released to all customers we can remove this endpoint and add expose this on patientDTO

Reply:
Were not exposing this to every customer yet and trying to avoid having to update all the schemas and sdks to get this out quicker

Reply:
What other places we would need to update beyond `PatientDTO`?

AFAIK that would be the only place, and we could even not update the SDK so only the Dash would be updated to “force” reading this property from the SDK response (we don’t do zod schema validation on the SDK for the `getPatient`).

If that’s true, that would be easier/faster than building a new endpoint now, plus we don’t have to refactor this in the future.